src/ameba.ts

```diff
@@ -29,13 +29,23 @@ export class Ameba {
         this.config = getConfig();
     }
 
-    public execute(document: TextDocument): void {
-        if (document.languageId !== 'crystal' || document.isUntitled || document.uri.scheme !== 'file') {
-            return;
+    public execute(document: TextDocument, virtual: boolean = false): void {
+        if (!isCrystalDocument(document)) return;
+        if (isDocumentVirtual(document) && !virtual) return;
+
+        const dir = (workspace.getWorkspaceFolder(document.uri) ?? noWorkspaceFolder(document.uri)).uri.fsPath;
+
+        const args = [this.config.command, '--format', 'json'];
+
+        if (!virtual) {
+            args.push(document.fileName)
+        } else {
+            args.push('--stdin-filename', document.fileName);
+
+            // Disabling these as they're common when typing
+            args.push('--except', 'Lint/Formatting,Layout/TrailingBlankLines,Layout/TrailingWhitespace,Naming/Filename');
```

Comment:
```suggestion
            // Disabling these as they're common when typing
            args.push('--except', 'Lint/Formatting,Layout/TrailingBlankLines,Layout/TrailingWhitespace');

            args.push('-', document.fileName);
```

Reply:
I'd prefer to stick with `--stdin-filename` so I don't need to deal with `virtual` when parsing the results.

Also the Naming/Filename rule is necessary as untitled documents will flag that.

Reply:
I see. In any case that's not really necessary since you already know the path given you're running `execute` on a single file here.